jib-core/src/test/java/com/google/cloud/tools/jib/cache/RetryTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.jib.cache;
+
+import java.util.concurrent.TimeUnit;
+import org.junit.Assert;
+import org.junit.Test;
+
+/** Tests for {@link Retry}. */
+public class RetryTest {
+  private volatile int actionCount = 0;
+  private final Retry.Action<Exception> countingAction =
+      () -> {
+        synchronized (this) {
```

Comment:
I don't think we really need synchronization. There is no additional thread and every code is executed in the main thread. Even if a thread was involved, we can use `LongAdder` or `AtomicInteger.incrementAndGet()` (both under `java.util.concurrent.atomic`?

Reply:
You do... If it's volatile. But it's a good point that since it's on the same thread then volatile isn't required.